.github/workflows/code-quality.yml

```diff
@@ -63,3 +63,79 @@ jobs:
         with:
           local-ref: ${{ github.event.pull_request.head.sha }}
           remote-ref: origin/${{ github.event.pull_request.base.ref }}
+
+  notebook-validation:
```

Comment:
Will this actually run all of the notebooks in the repo? How will this handle notebooks that require specialized hardware or take an extended time to run?